schema_generator/search.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import print_function
+
+import os.path
+import sys
+import json
+from collections import defaultdict, OrderedDict
+
+from dmcontent import ContentLoader, utils
+
+
+_base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+def _get_questions_by_type(framework_slug, question_types):
+    loader = ContentLoader(_base_dir)
+    loader.load_manifest(
+        framework_slug,
+        'services',
+        'search_filters'
+    )
+
+    manifest = loader.get_manifest(framework_slug, 'search_filters')
+    return (q for q in sum((s.questions for s in manifest.sections), []) if q.type in question_types)
+
+
+def _checkbox_tree_transformation_generator(checkbox_tree_question):
+    def update_ancestors_dict(options, leaf_values_dict, parents):
+        for option in options:
+            children = option.get('options', [])
+            if not children:
+                if parents:
+                    leaf_values_dict[parents].append(utils.get_option_value(option))
+            else:
+                update_ancestors_dict(children, leaf_values_dict, parents.union([utils.get_option_value(option)]))
+
+    leaf_values_by_ancestor_set = defaultdict(list)
+    # list of child-values preserves order from the source yaml, for the benefit of git history in output file
+
+    update_ancestors_dict(checkbox_tree_question.options, leaf_values_by_ancestor_set, parents=frozenset())
+
+    return [
+        OrderedDict((
+            ('field', checkbox_tree_question.id),
+            ('any_of', child_values),
+            ('append_value', sorted(ancestor_values)),
+        )) for ancestor_values, child_values in leaf_values_by_ancestor_set.items()
+        ]
+
+
+transformation_generators = {
```

Comment:
Sure, or should I give it a name starting with underscore? Probably not both I guess :)